TypeCobol/Compiler/SqlAnalysis/SqlASTChecker.cs

```diff
@@ -0,0 +1,36 @@
+﻿using JetBrains.Annotations;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TypeCobol.Analysis;
+using TypeCobol.Compiler.Diagnostics;
+using TypeCobol.Compiler.Nodes;
+using TypeCobol.Compiler.SqlAnalysis.ASTRules;
+using TypeCobol.Compiler.SqlCodeElements;
+using TypeCobol.Compiler.SqlNodes;
+
+namespace TypeCobol.Compiler.SqlAnalysis
+{
+    /// <summary>
+    /// SQL AST Checker class
+    /// </summary>
+    internal class SqlASTChecker : AbstractAstWithSqlVisitor
```

Comment:
To private analyzer project.

Reply:
Ok it will be a task, this POC is not the final result, only concept should be validated here

Reply:
This is a POC but the goal is also to merge this PR in order to have a canvas for all further SQL issues.
So this class can be removed from this PR. We have seen that it's working.

Reply:
done in [7f2afb7](https://github.com/TypeCobolTeam/TypeCobol/pull/1928/commits/7f2afb778747d8f26ff0ce751781c2ab8845aa78)